sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/pixels/SyncPixels.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.impl.pixels
+
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.sync.impl.pixels.SyncPixelValues.Feature
+import com.duckduckgo.sync.impl.stats.SyncStatsRepository
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface SyncPixels {
+    fun fireStatsPixel()
+
+    fun fireMergeConflictPixel(feature: Feature)
+
+    fun fireOrphanPresentPixel(feature: Feature)
+    fun fireEncryptFailurePixel()
+
+    fun fireDecryptFailurePixel()
+
+    fun fireCountLimitPixel(feature: Feature)
+}
+
+@ContributesBinding(AppScope::class)
+class RealSyncPixels @Inject constructor(
+    private val pixel: Pixel,
+    private val statsRepository: SyncStatsRepository,
+) : SyncPixels {
+    override fun fireStatsPixel() {
+        val dailyStats = statsRepository.getDailyStats()
+        pixel.fire(
+            SyncPixelName.SYNC_SUCCESS_RATE,
+            mapOf(
+                SyncPixelParameters.RATE to dailyStats.successRate.toString(),
+            ),
+        )
+        pixel.fire(
+            SyncPixelName.SYNC_DAILY_ATTEMPTS,
+            mapOf(
+                SyncPixelParameters.ATTEMPTS to dailyStats.attempts.toString(),
+            ),
+        )
+    }
+
+    override fun fireMergeConflictPixel(feature: Feature) {
```

Comment:
Proposal to accept `String` instead of `Feature`, so we can directly use `SyncableType.field`.